dlls/winebus.sys/bus_udev.c

```diff
@@ -1701,28 +1701,27 @@ static void udev_add_device(struct udev_device *dev, int fd)
         memcpy(desc.serialnumber, zeros, sizeof(zeros));
     }
 
-    if (!is_hidraw_enabled(desc.vid, desc.pid, axes, buttons))
-    {
-        TRACE("hidraw %s: deferring %s to a different backend\n", debugstr_a(devnode), debugstr_device_desc(&desc));
-        close(fd);
-        return;
-    }
-    if (is_sdl_blacklisted(desc.vid, desc.pid))
-    {
-        /* this device is blacklisted */
-        TRACE("hidraw %s: ignoring %s, in SDL blacklist\n", debugstr_a(devnode), debugstr_device_desc(&desc));
-        close(fd);
-        return;
-    }
 #ifdef HAS_PROPER_INPUT_HEADER
-    else
-        desc.is_gamepad = (axes == 6 && buttons >= 14);
+    desc.is_gamepad = (axes == 6 && buttons >= 14);
```

Comment:
This seems problematic because this will now define devices as gamepad that have 6 axis and more than 14 buttons - like common HOTAS do have. This has been explicitly in the `else` case so `is_hidraw_enabled` can override this behavior.

Reply:
Looking at this again, the `else` probably didn't make sense in the first place because both `if` return from the function.

Reply:
Yes. That is why I took it out.

I see what you mean about HOTAS. This seems to be a standard heuristics used in all the input types though ([iohid](https://github.com/ValveSoftware/wine/blob/d8766d0/dlls/winebus.sys/bus_iohid.c#L340) and [sdl](https://github.com/ValveSoftware/wine/blob/d8766d0/dlls/winebus.sys/bus_sdl.c#L1051)). Probably the only way to really get it right is to not rely on a heuristics at all if [hwdb](https://www.freedesktop.org/software/systemd/man/latest/hwdb.html) info is available.

The hwdb info can already be extracted via the [`what_am_I`](https://github.com/ValveSoftware/wine/blob/d8766d0/dlls/winebus.sys/bus_udev.c#L870) function. It just isn't being used here. Maybe it should be?

Reply:
Yeah, SDL has a similar problem of identifying some joysticks as gamepads... But because axis centers differ, this creates phantom movement. So we need to ensure such devices do not become mapped to the xinput driver.

Reply:
I think I can mark this as resolved?